src/index/VocabularyGenerator.cpp

```diff
@@ -0,0 +1,125 @@
+#include "./VocabularyGenerator.h"
+#include <unordered_set>
+#include <vector>
+#include <queue>
+#include <string>
+#include <utility>
+#include <fstream>
+#include <iostream>
+
+#include "../util/Log.h"
+
+class PairCompare {
+ public:
+  bool operator()(const std::pair<std::string, size_t>& p1, const std::pair<std::string, size_t>& p2) {
+    return p1.first > p2.first;}
+};
+// ___________________________________________________________________
+void mergeVocabulary(const std::string& basename, size_t numFiles) {
+  size_t bufferSize = 200000;
+  std::vector<std::fstream> infiles;
+  std::vector<std::vector<std::string>> vecs(numFiles);
+  std::ofstream outfile(basename + ".vocabulary");
+  std::ofstream outfileExternal(basename + "externalTextFile");
+  std::vector<bool> endOfFile(numFiles, false);
+  std::vector<std::streampos> posNextWord;
+  std::vector<std::vector<std::string>::iterator> iterators;
```

Comment:
This could also be named more tellingly, these aren't just iterators, they are the current position for each file, right? So maybe `currPositions`